explorer/interpreter/resolve_names.cpp

```diff
@@ -413,6 +413,17 @@ static auto ResolveNames(Statement& statement, StaticScope& enclosing_scope)
       CARBON_RETURN_IF_ERROR(ResolveNames(while_stmt.body(), enclosing_scope));
       break;
     }
+    case StatementKind::For: {
+      StaticScope block_scope;
```

Comment:
This scope doesn't really correspond to a block, right? It corresponds to the whole `for` statement. That being the case, maybe this should be called something like `statement_scope`?

Reply:
Yes, it is a better name!